projects/omang-service-mediator/src/modules/omang/services/omang.service.ts

```diff
@@ -0,0 +1,263 @@
+      // Resource Type
+      fhirPatient.resourceType = 'Patient';
+      //Id
+      fhirPatient.id = omang.IdNo;
+
+      // Identifier
+      const pat_identifier = new fhirR4.Identifier();
+      pat_identifier.system = config.get('ClientRegistry:OmangSystem');
+      pat_identifier.value = omang.IdNo;
+
+      // Hash Unique Internal ID
+      const hashedId = calculateMD5Hash(omang.IdNo);
+      const internal_identifier = new fhirR4.Identifier();
+      internal_identifier.system =
+        'http://omang.bw.org/ext/identifier/internalid';
+      internal_identifier.value = hashedId;
+      fhirPatient.identifier = [pat_identifier, internal_identifier];
+
+      //Active
+      fhirPatient.active = true;
+      // Name
+      const pat_name = new fhirR4.HumanName();
+      pat_name.family = omang.Surname;
+      pat_name.given = omang.FirstName.split(' ');
+      fhirPatient.name = [pat_name];
+
+      // Gender
+      switch (omang.Sex) {
+        case 'F':
+          fhirPatient.gender = fhirR4.Patient.GenderEnum.Female;
+          break;
+        case 'M':
+          fhirPatient.gender = fhirR4.Patient.GenderEnum.Male;
+          break;
+      }
+
+      // Birthdate
+      fhirPatient.birthDate = omang.BirthDate?.toISOString().slice(0, 10);
+
+      // Deceased
+      if (omang.DeceasedDate) {
+        fhirPatient.deceasedDateTime = omang.DeceasedDate.toISOString().slice(
+          0,
+          10,
+        );
+      }
+
+      // Address
+      const address = new fhirR4.Address();
+      address.district = omang.DistrictName;
+      address.postalCode = omang.DistrictCode;
+
+      fhirPatient.address = [address];
+
+      // Marital Status
+
+      const system_url = 'http://hl7.org/fhir/R4/valueset-marital-status.html';
+      fhirPatient.maritalStatus = new fhirR4.CodeableConcept();
+      fhirPatient.maritalStatus.coding = [];
+      const theCoding = new fhirR4.Coding();
+      theCoding.system = system_url;
+      switch (omang.MaritalStatusCode) {
+        case 'MAR':
+          theCoding.code = 'M';
+          break;
+        case 'SGL':
+          theCoding.code = 'S';
+          break;
+        case 'WDW':
+          theCoding.code = 'W';
+          break;
+        case 'DIV':
+          theCoding.code;
+          break;
+        case 'SEP':
+          theCoding.code;
+          break;
+        case 'WHD':
+          theCoding.code = 'UNK';
+          break;
+      }
+
+      fhirPatient.maritalStatus.coding.push(theCoding);
+    }
+
+    return fhirPatient;
+  }
+
+  private mapOmangToSearchBundle(omangRecords: Omang[]): fhirR4.Bundle {
+    const searchBundle: fhirR4.Bundle = FhirAPIResponses.RecordInitialized;
+
+    for (const omang of omangRecords) {
+      const patient: fhirR4.Patient = this.mapOmangToFhirPatient(omang);
+
+      const entry = new fhirR4.BundleEntry();
+      entry.fullUrl =
+        config.get('ClientRegistry:OmangSystem') +
+        patient.constructor.name +
+        patient.id;
```

Comment:
Yup we may need to double check this, it seems missing slashes

Reply:
In the former Omang implementation, they do exactly the same thing without slashes.

Reply:
Ok cool, I think then it makes sense to keep this as it was done in the original omang service. Lets just keep this in mind in case we notice any issues with incorrect fullUrl being created :+1: